core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -294,17 +294,27 @@ public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.Part
 
                 futures.put(topicIdPartition, future);
             } else {
-                futures.put(topicIdPartition, CompletableFuture.completedFuture(Errors.UNKNOWN_TOPIC_OR_PARTITION));
+                futures.put(topicIdPartition, CompletableFuture.completedFuture(Errors.UNKNOWN_TOPIC_OR_PARTITION.exception()));
             }
         });
 
         CompletableFuture<Void> allFutures = CompletableFuture.allOf(
             futures.values().toArray(new CompletableFuture[0]));
         return allFutures.thenApply(v -> {
             Map<TopicIdPartition, ShareAcknowledgeResponseData.PartitionData> result = new HashMap<>();
-            futures.forEach((topicIdPartition, future) -> result.put(topicIdPartition, new ShareAcknowledgeResponseData.PartitionData()
-                .setPartitionIndex(topicIdPartition.partition())
-                .setErrorCode(future.join().code())));
+            futures.forEach((topicIdPartition, future) -> {
+                ShareAcknowledgeResponseData.PartitionData partitionData = new ShareAcknowledgeResponseData.PartitionData()
+                        .setPartitionIndex(topicIdPartition.partition());
+                Throwable t = future.join();
+                if (t == null) {
+                    partitionData.setErrorCode(Errors.NONE.code())
+                            .setErrorMessage(Errors.NONE.message());
```

Comment:
Why to set a error message in case of null?

Reply:
ditto